src/twisted/internet/defer.py

```diff
@@ -1041,7 +1041,9 @@ def maybeSucceed(result: object) -> None:
         return future
 
     @classmethod
-    def fromFuture(cls, future: Future) -> "Deferred[Any]":
+    def fromFuture(
+        cls, future: "Future[_DeferredResultT]"
+    ) -> "Deferred[_DeferredResultT]":
```

Comment:
Now I'm very confused about `_DeferredResultT` being contravariant... :/

Reply:
It's definitely wrong, but having spent some time trying to fix this now, it becomes clear that the real signatures for Deferred's methods are fiendishly complex, and this contravariance papers over some of the complexities (by accepting invalid types).

Trying to see if I can golf my way to a successful replacement that *doesn't* do that.